### PR TITLE
Feat: Integration iugu pix payment payment

### PR DESCRIPTION
**Pull request by femacedogomes**

### CODE REVIEW

File: src/utils/copyTextToClipboard.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const copyTextToClipboard = (value: string) => {
+  const tempTextArea = document.createElement("textarea");
+  tempTextArea.value = value;
+  document.body.appendChild(tempTextArea);
+  tempTextArea.select();
+  document.execCommand("copy");
+  document.body.removeChild(tempTextArea);
+
+  alert("Texto copiado para a área de transferência!");
```
Comment (pedrohbl03) [inline src/utils/copyTextToClipboard.ts:9]: Alert?

Change this to a toast?

u can remove the line 9 and set a toastify after copyTextToClipboard has been executed.

Comment (pedrohbl03) [inline src/utils/copyTextToClipboard.ts:9]:
```suggestion
export const copyTextToClipboard = (value: string) => {
  const tempTextArea = document.createElement("textarea");
  tempTextArea.value = value;
  document.body.appendChild(tempTextArea);
  tempTextArea.select();
  document.execCommand("copy");
  document.body.removeChild(tempTextArea);
```

File: src/app/app/(application)/pagamento/WaitingApprovalStep.tsx [hunk 2]
```diff
@@ -1,19 +1,74 @@
...
+        }
+      })
+      .catch(() => {
+        showToastify({
+          label: "Ocorreu um erro ao atualizar dados do pagamento",
+          type: "error",
+        });
+      });
+  };
+
```
Comment (pedrohbl03) [inline src/app/app/(application)/pagamento/WaitingApprovalStep.tsx:71]:
```suggestion
 const handleClickCopyText = (textToCopy) => {
     try {
        copyTextToClipboard(textToCopy);
         showToastify({
          label: "Texto copiado com sucesso para area de transferencia",
          type: "success",
        });
     } catch (error) {
         showToastify({
          label: "Falha ao copiar texto para aréa de transferência.",
          type: "error",
        });
     }
 }
```

File: src/app/app/(application)/pagamento/WaitingApprovalStep.tsx [hunk 3]
```diff
@@ -24,23 +79,27 @@ export const WaitingApprovalStep: React.FC<IWaitingApprovalStepProps> = ({
...
               <p className="text-xs max-w-[280px] mb-2 break-words md:max-w-[340px]">
-                00020101021226990014br.gov.bcb.pix2577pix.bpp.com.br/14796606/qrdaksldkalsdkamdl,KDl123
-                KmIYgzfr3AZ38E8vQbnYv6xqDPenH0KehYAyWeXfjF5204000053039865802BR591{" "}
+                {pixData.qrCodeValue.text}
               </p>
             </div>
 
-            <ButtonFourth className="!border-generic-limeGreen !border-[1px] !py-1 !text-xs  !text-generic-limeGreen">
+            <ButtonFourth
+              onClick={() => copyTextToClipboard(pixData.qrCodeValue.text)}
```
Comment (pedrohbl03) [inline src/app/app/(application)/pagamento/WaitingApprovalStep.tsx:100]:
```suggestion
              onClick={() => handleClickCopyText(pixData.qrCodeValue.text)}
```

**Review by pedrohbl03 — CHANGES_REQUESTED** [review]

File: src/app/app/(application)/pagamento/page.tsx [hunk 4]
```diff
@@ -15,19 +16,33 @@ const PaymentPage = () => {
...
+      text: "",
+    },
+  });
   const [paymentTab, setPaymentTab] = useState(PAYMENT.creditCard);
   const renderStep = (step: number) => {
     switch (step) {
       case STEPS.PAYMENT:
         return (
           <PaymentStep
             PaymentTab={paymentTab}
```
[the remark this answers is not in the document]
Comment (femacedogomes) [reply src/app/app/(application)/pagamento/page.tsx]: done

File: src/app/app/(application)/pagamento/page.tsx [hunk 5]
```diff
@@ -15,19 +16,33 @@ const PaymentPage = () => {
...
+            setPaymentTab={setPaymentTab}
             setCurrentStep={setCurrentStep}
+            setPixData={setPixData}
           />
         );
       case STEPS.LOADING_PAYMENT:
-        return <WaitingApprovalStep PaymentTab={paymentTab} />;
+        return (
+          <WaitingApprovalStep
+            PaymentTab={paymentTab}
```
[the remark this answers is not in the document]
Comment (femacedogomes) [reply src/app/app/(application)/pagamento/page.tsx]: done

File: src/app/app/(application)/pagamento/page.tsx [hunk 6]
```diff
@@ -36,8 +51,9 @@ const PaymentPage = () => {
         return (
           <PaymentStep
             PaymentTab={paymentTab}
```
[the remark this answers is not in the document]
Comment (femacedogomes) [reply src/app/app/(application)/pagamento/page.tsx]: done

File: src/utils/copyTextToClipboard.ts (hunk 1 above)
Comment (femacedogomes) [reply src/utils/copyTextToClipboard.ts:9]: done

Comment (femacedogomes) [reply src/utils/copyTextToClipboard.ts]: done

File: src/app/app/(application)/pagamento/WaitingApprovalStep.tsx (hunk 2 above)
Comment (femacedogomes) [reply src/app/app/(application)/pagamento/WaitingApprovalStep.tsx:86]: done

File: src/app/app/(application)/pagamento/WaitingApprovalStep.tsx (hunk 3 above)
Comment (femacedogomes) [reply src/app/app/(application)/pagamento/WaitingApprovalStep.tsx]: done

**Pull request merged**